curriculum/challenges/english/25-front-end-development/lab-project-idea-board/67051431a73f1ca25d9a6f25.md

```diff
@@ -0,0 +1,334 @@
+---
+id: 67051431a73f1ca25d9a6f25
+title: Build a Project Idea Board
+challengeType: 14
+dashedName: build-a-project-idea-board
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should define an object constant named `projectStatus` with the following project statuses and descriptions:
+
+   - `PENDING` with a description of `Pending Execution`.
+   - `SUCCESS` with a description of `Executed Successfully`.
+   - `FAILURE` with a description of `Execution Failed`.
+
+2. You should define a class named `ProjectIdea` with a `constructor` that takes a `title` and a `description` as parameters and initialize the `title` and `description` properties based on the parameters passed. The class should also have a property named `status` and be set to the description of the `PENDING` status.
+3. You should define a method named `updateProjectStatus` inside the `ProjectIdea` class that takes accepts `newStatus` as parameter and can update the project idea status.
+4. You should define a `ProjectIdeaBoard` class with a `constructor` that accepts a `title` and initializes an empty array named `ideas` to hold instances of the `ProjectIdea` class.
+5. You should define a method named `pin` inside the `ProjectIdeaBoard` class that accepts an instance of the `ProjectIdea` class and pushes the given instance to the `ideas` array.
+6. You should define a method named `unpin` inside the `ProjectIdeaBoard` class that accepts an instance of the `ProjectIdea` class and removes the given instance from the `ideas` array.
+7. You should define a method named `count` that returns the number of project ideas in the given `ProjectIdeaBoard` array.
+8. You should define a method named `toString` that returns the name of the projects ideas, their description and status in the format:
+
+```js
+[ProjectIdeaBoard title] has [ProjectIdeaBoard count] idea(s)\n
+[ProjectIdea Title] ([ProjectIdea Status Description]) - [ProjectIdea Description]
+[ProjectIdea Title] ([ProjectIdea Status Description]) - [ProjectIdea Description]
```

Comment:
I remember we agreed on using square brackets for placeholders, like step#2 [here](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md).

Reply:
In any case we are using both types of brackets https://github.com/freeCodeCamp/freeCodeCamp/blob/4cc7c54979f5c7715603648f4f497223a87e168f/curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md?plain=1#L43

I prefer angular brackets but it would be okay as long as we are consistent. I would like to hear other thoughts on this.

Reply:
I think angular brackets are not a symbol for data types in JS so it is less confusing

Reply:
I have used angular brackets now.